apps/core/components/product-sheet/index.tsx

```diff
@@ -1,33 +1,52 @@
 'use client';
 
-import { Button } from '@bigcommerce/components/button';
-import { Rating } from '@bigcommerce/components/rating';
 import {
   Sheet,
   SheetClose,
   SheetContent,
   SheetHeader,
   SheetOverlay,
   SheetTitle,
-  SheetTrigger,
 } from '@bigcommerce/components/sheet';
-import { Loader2 as Spinner } from 'lucide-react';
-import Image from 'next/image';
-import { useSearchParams } from 'next/navigation';
-import { PropsWithChildren, useEffect, useId, useState } from 'react';
+import { usePathname, useRouter, useSearchParams } from 'next/navigation';
+import { useEffect, useState } from 'react';
 
-import { getProduct } from '~/client/queries/get-product';
-import { ProductForm } from '~/components/product-form';
-import { cn } from '~/lib/utils';
+import { ProductSheetContent } from './product-sheet-content';
+
+export const ProductSheet = () => {
+  const router = useRouter();
+  const pathname = usePathname();
+  const searchParams = useSearchParams();
+  const productId = searchParams.get('showQuickAdd');
 
-export const ProductSheet = ({ children, title }: PropsWithChildren<{ title: string }>) => {
   const [open, setOpen] = useState(false);
 
+  useEffect(() => {
+    setOpen(!!productId);
+  }, [productId]);
```

Comment:
We can derive `open` by doing something like:

```
const open = !!productId;
```

No need for this effect and state.

Reply:
Unfortunately there is a hydration issue with this approach. I believe it could be related to something to do with the searchParams when rendering, or an issue with the state directly in the radix components. Will merge as is for now and we can investigate other solutions at a later point.